src/NuGet.Clients/NuGet.PackageManagement.UI/ViewModels/ReadMePreviewViewModel.cs

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.VisualStudio.Telemetry;
+
+namespace NuGet.PackageManagement.UI.ViewModels
+{
+    public sealed class ReadMePreviewViewModel : ViewModelBase
+    {
+        public ReadMePreviewViewModel()
+        {
+            _isReadMeAvailable = false;
+            _isErrorWithReadMe = false;
+            _rawReadMe = string.Empty;
+        }
+
+        private bool _isErrorWithReadMe;
+
+        public bool IsErrorWithReadMe
+        {
+            get => _isErrorWithReadMe;
+            set
+            {
+                _isErrorWithReadMe = value;
+                RaisePropertyChanged(nameof(IsErrorWithReadMe));
+            }
+        }
+
+        private bool _isReadMeAvailable;
+
+        public bool IsReadMeAvailable
+        {
+            get => _isReadMeAvailable;
+            set
+            {
+                _isReadMeAvailable = value;
+                RaisePropertyChanged(nameof(IsReadMeAvailable));
+            }
+        }
+
+        private string _rawReadMe;
+
+        public string ReadMeMarkdown
+        {
+            get => _rawReadMe;
+            set
+            {
+                _rawReadMe = value;
+                RaisePropertyChanged(nameof(ReadMeMarkdown));
+            }
+        }
+
+        public async Task UpdateReadMe(string packagePath, string packageId)
+        {
+            var newReadMeValue = string.Empty;
+            var isReadMeAvailable = false;
+            var isErrorWithReadMe = false;
+            try
+            {
+                if (!string.IsNullOrEmpty(packagePath))
+                {
+                    var fileInfo = new FileInfo(packagePath);
+                    if (fileInfo.Exists)
+                    {
+                        using var stream = fileInfo.OpenRead();
+                        using var archive = new ZipArchive(stream, ZipArchiveMode.Read);
+                        var nuspecZipArchiveEntry = archive.Entries.FirstOrDefault(zipEntry => string.Equals(UnescapePath(zipEntry.Name), $"{packageId}{PackagingCoreConstants.NuspecExtension}", StringComparison.OrdinalIgnoreCase));
+                        if (nuspecZipArchiveEntry is not null)
```

Comment:
The else here is that the README wasn't able to be read, but I don't see how the `isErrorWithReadMe` would become true.
I think the rendering would show no README, but shouldn't it show an error? At least it'd be good to fire telemetry so we can see how common it is that we can't read the archives.

Reply:
So the else here would mean that a nuspec wasn't found in the nupkg, which would be weird but I didn't see it as a problem specifically with rendering the readme information and with the package itself. 

I think logging telemetry for it makes sense since it's a strange scenario, I was more so trying to avoid a null reference exception.